### ISSUE TITLE
Notification management

### ISSUE THREAD

**Issue by imacrayon** [imacrayon/snowbodyknows#42]
The app now sends emails when wishes are added to a wishlist and when comments are posted to a wishlist. There should probably be an area where users can customize what kinds of email notifications they want to receive.

**Comment by sebcar** [comment]
Hi! Is someone working on this?

**Comment by sebcar** [comment]
Hi! Sorry for the delay.

![image](https://github.com/imacrayon/snowbodyknows/assets/11927587/eea92f92-2a5a-4c29-8a70-b64b7db6b60e)

Is this ok in the profile update page?

**Comment by imacrayon** [comment]
Awesome, that looks great!

**Comment by sebcar** [comment]
It is ok to have a Configuration Model with `id, value, user_id` to manage different configuration options in the future?

**Comment by imacrayon** [comment]
Sure, I think that’s fine. I wouldn’t be opposed to using a JSON column either. Whatever you prefer is cool.

**Comment by sebcar** [comment]
A JSON column sounds good, so it will be something like `user_id, value(json)`

**Comment by imacrayon** [comment]
Sorry, I meant a JSON column on the `users` table called `settings` or something. Instead of making a whole new table. If you’ve got a lot of ideas for setting though, maybe a separate table would be the way to go?

**Comment by sebcar** [comment]
I'm going for a settings column in users :)

**Comment by sebcar** [comment]
Everything working except the notification test
![image](https://github.com/imacrayon/snowbodyknows/assets/11927587/0bbb07af-1007-4327-9473-efab7ec0d7f7)
Cannot change the settings value for the `$viewer`

**Comment by imacrayon** [comment]
I would double check that you have `WishCreatedNotification` imported in your test, sometimes I forget to do that. If that's not the cause try doing:
```php
Notification::assertNotSentTo($viewer, WishCreatedNotification::class);
```
See if that passes. I've has issues with using a closure with `assertNotSentTo` in the past.

**Comment by imacrayon** [comment]
Line 29 in `WishController` is kind of hard to follow. It might be easier to read if you used `filter` instead of `reject`:
```php
Notification::send($withlist->viewers->filter(fn ($viewer) => $viewer->settings['notification']['wish-created']), ...);
```

**Comment by sebcar** [comment]
Ok, let's fix that
